src/web/app/components/visibility-panel/visibility-panel.component.html

```diff
@@ -0,0 +1,57 @@
+<div class="margin-top-15px padding-15px" [ngClass]="{ 'background-color-light-green': !model.isQuestionHasResponses, 'alert alert-danger': model.isQuestionHasResponses }">
+  <div>
+    <div *ngIf="model.isQuestionHasResponses">
+      <h5>Changing the visibility after collecting responses is not recommended.</h5>
+      <p>Reason: The existing responses were submitted under the 'promise' of a certain visibility and changing the visibility later 'breaks' that promise.</p>
+    </div>
+    <b [ngClass]="{ 'visibility-title': !model.isQuestionHasResponses }">Visibility</b> (Who can see the responses?)
+  </div>
+  <div ngbDropdown class="margin-top-15px">
+    <button id="btn-question-visibility" class="btn btn-light white-space-normal" ngbDropdownToggle [disabled]="!model.isEditable">
+      <span *ngIf="!model.isUsingOtherVisibilitySetting">
+        {{ model.commonVisibilitySettingName }}
+      </span>
+      <span *ngIf="model.isUsingOtherVisibilitySetting">Custom visibility options</span>
+    </button>
+    <ul id="question-visibility-dropdown" ngbDropdownMenu>
+      <li class="dropdown-header">Common visibility options</li>
+      <li class="dropdown-item" *ngFor="let visibilityOption of commonFeedbackVisibilitySettings"
+          (click)="applyCommonVisibilitySettings(visibilityOption)">{{ visibilityOption.name }}</li>
+      <li class="dropdown-divider" *ngIf="isCustomFeedbackVisibilitySettingAllowed"></li>
+      <li class="dropdown-item" *ngIf="isCustomFeedbackVisibilitySettingAllowed" (click)="triggerCustomVisibilitySetting()">Custom visibility options...</li>
+    </ul>
+  </div>
+  <table class="table margin-top-15px background-color-white table-striped" *ngIf="model.isUsingOtherVisibilitySetting">
+    <thead>
+    <tr>
+      <th scope="col" class="text-left">User/Group</th>
+      <th scope="col" class="text-center" *ngFor="let visibilityControl of visibilityControls | enumToArray">{{ visibilityControl | visibilityControlName }}</th>
+    </tr>
+    </thead>
+    <tbody id="custom-visibility-table">
+    <ng-container *ngFor="let visibilityType of feedbackVisibilityTypes | enumToArray">
```

Comment:
Similarly here for `feedbackVisibilityTypes` vs `FeedbackVisibilityType`

Reply:
fixed in 850dcc8